platforms/core-execution/request-handler-worker/build.gradle.kts

```diff
@@ -13,23 +13,25 @@ dependencies {
     api(projects.baseServices)
     api(projects.buildOperations)
     api(projects.concurrent)
+    api(projects.hashing)
+    // TODO: Ideally, we would not depend on model-core in a worker.
+    // All we really want is the instantiation infrastructure, but this
+    // brings in core-api, which should be avoided in workers.
+    api(projects.modelCore)
     api(projects.messaging)
+    api(projects.problemsApi)
     api(projects.serialization)
+    api(projects.snapshotsWorker)
+    api(projects.stdlibJavaExtensions)
     api(projects.workerMain)
 
     implementation(projects.classloaders)
+    implementation(projects.coreApi)
     implementation(projects.persistentCache)
-    implementation(projects.problemsApi)
     implementation(projects.serviceLookup)
     implementation(projects.serviceProvider)
     implementation(projects.serviceRegistryBuilder)
-    implementation(projects.stdlibJavaExtensions)
-
-    // TODO: Ideally, we would not depend on model-core in a worker.
-    // All we really want is the instantiation infrastructure, but this
-    // brings in core-api, which should be avoided in workers.
-    implementation(projects.modelCore)
+    implementation(projects.snapshots)
```

Comment:
It is very fishy to me that we created this `snapshots-worker` project, but still depend on the non-worker snapshots project in worker code.